app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -378,6 +380,27 @@ class DefaultSessionControlController(
             metrics.track(Event.PocketTopSiteClicked)
         }
 
+        val baseURL = url.tryGetHostFromUrl()
+        if (SupportUtils.GOOGLE_URL.contains(baseURL)) {
+            val availableEngines = activity
+                .components
+                .core
+                .store
+                .state
+                .search
+                .searchEngines
+
+            val searchAccessPoint = Event.PerformedSearch.SearchAccessPoint.TOPSITE
+            val event =
+                availableEngines.firstOrNull { engine -> engine.suggestUrl?.contains(baseURL) == true }
+                    ?.let { searchEngine ->
+                        searchAccessPoint.let { sap ->
+                            MetricsUtils.createSearchEvent(searchEngine, store, sap)
+                        }
+                    }
+            event?.let { activity.metrics.track(it) }
```

Comment:
We should be able to add a test for this in SessionControlControllerTest. There should be something there already for metrics.track(Event.TopSiteOpenGoogle) as well.